src/sonic-yang-models/yang-models/sonic-port.yang

```diff
@@ -63,6 +63,44 @@ module sonic-port{
 					}
 				}
 
+				leaf autoneg {
+					description "Port auto negotiation mode";
+
+					type string {
+						pattern "on|off";
+					}
+				}
+				
+				leaf-list adv_speeds {
+					description "Port advertised speeds, valid value could be a list of interger or all";
+
+					type union {
+						type uint32 {
+							range 1..400000;
+						}
+						type string {
+							pattern "all";
+						}
+					}
+				}
+
+				leaf interface_type {
+					description "Port interface type";
+
+					type stypes:interface_type;
+				}
+
+				leaf-list adv_interface_types {
+					description "Port advertised interface type, valid value could be a list of stypes:interface_type or all";
```

Comment:
Kindly point to the correct paragraph in HLD here in comment, thanks again

Reply:
https://github.com/Junchao-Mellanox/SONiC/blob/port-auto-neg/doc/port_auto_neg/port-auto-negotiation-design.md#config-advertised-interface-types

Reply:
I cannot add the HLD paragraph to the comment here because HLD is not approved yet. And HLD approval requires YANG model PR approval.